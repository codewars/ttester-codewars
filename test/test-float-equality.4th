\ Copyright 2023 nomennescio

\ pass
<{ 1e -> 1e }>
\ fail
<{ 1.001e -> 1e }>

\ abs<> ( f1 f2 -- ? ) := |f1-f2| > epsilon
F<>: abs<>
0.01e epsilon f!
\ pass
<{ 1.001e -> 1e }>
\ fail
<{ 1.1e -> 1e }>
1e epsilon f!
\ pass
<{ 1.1e -> 1e }>

\ rel<> ( f1 f2 -- ? ) := |f1-f2|/|f1+f2| > epsilon
F<>: rel<>
0.05e epsilon f!
\ pass
<{ 5.5e -> 5e }>
<{ 11e -> 10e }>
\ fail
<{ 5.6e -> 5e }>
<{ 11.2e -> 10e }>

\ f<> ( f1 f2 -- ? ) := f1 <> f2
F<>: f<>
\ pass
<{ 1e -> 1e }>
\ fail
<{ 1.001e -> 1e }>
