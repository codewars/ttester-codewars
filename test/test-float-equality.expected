
<PASSED::>Test Passed

<FAILED::>Expected 1. , got 1.001 

<PASSED::>Test Passed

<FAILED::>Expected 1. , got 1.1 

<PASSED::>Test Passed

<PASSED::>Test Passed

<PASSED::>Test Passed

<FAILED::>Expected 5. , got 5.6 

<FAILED::>Expected 10. , got 11.2 

<PASSED::>Test Passed

<FAILED::>Expected 1. , got 1.001 
